src/Compiler/Reports/SynthesisReportManager.cpp

```diff
@@ -0,0 +1,136 @@
+
+#include <QFile>
+#include <QTextStream>
+
+#include "CompilerDefines.h"
+#include "NewProject/ProjectManager/project.h"
+#include "TableReport.h"
+#include "Utils/FileUtils.h"
+
+namespace {
+static constexpr const char *STAT_STR{"Printing statistics"};
+static constexpr const char *DE_STR{"DE:"};
+static constexpr const char *MAX_LVL_STR{"Maximum logic level"};
+static constexpr const char *AVG_LVL_STR{"Average logic level"};
+}  // namespace
+
+namespace FOEDAG {
+
+std::vector<std::string> SynthesisReportManager::getAvailableReportIds() const {
+  return {"Synthesis report"};
+}
+
+SynthesisReportManager::LinesData SynthesisReportManager::getStatistics(
+    QTextStream &in) const {
+  auto res = LinesData{};
+
+  auto line = QString{};        // Unmodified line from the log file
+  auto dataLine = QString{};    // Simplified line
+  auto parentItem = QString{};  // Parent item for lines starting with tab
+
+  in.readLineInto(&line);  // skip empty line
+  in.readLineInto(&line);  // skip top level module
+  in.readLineInto(&line);  // skip empty line
+
+  while (in.readLineInto(&line)) {
+    if (line.startsWith("     ") && parentItem.isEmpty())
+      parentItem = dataLine.left(dataLine.indexOf(':'));
+
+    dataLine = line.simplified();
+    if (dataLine.isEmpty()) break;
+
+    auto reportLine = std::vector<std::string>{};
+    auto lineStrs = dataLine.contains(":") ? dataLine.split(":")
+                                           : dataLine.split(QChar::Space);
+
+    for (auto i = 0; i < lineStrs.size(); ++i) {
+      auto lineStr = lineStrs[i];
+      // The first string represents statistic name. Parent item should only be
+      // added there.
+      if (!i && !parentItem.isEmpty())
+        lineStr = QString("%1:%2").arg(parentItem).arg(lineStr);
+      reportLine.push_back(lineStr.toStdString());
+    }
+
+    res.push_back(std::move(reportLine));
+  }
+
+  return res;
+}
+
+SynthesisReportManager::LinesData SynthesisReportManager::getLevels(
+    const QString &line) const {
+  auto res = LinesData{};
+
+  auto lineData = line.split(" ");
+  res.push_back({MAX_LVL_STR, lineData[10].toStdString()});
+  res.push_back({AVG_LVL_STR, lineData[14].toStdString()});
+
+  return res;
+}
+
+std::unique_ptr<ITaskReport> SynthesisReportManager::createReport(
+    const std::string &reportId) {
+  auto logPath =
+      std::filesystem::path(Project::Instance()->projectPath().toStdString()) /
+      std::string(SYNTHESIS_LOG);
+
+  if (!FileUtils::FileExists(logPath)) return nullptr;
+
+  auto pathString = QString::fromStdString(logPath.string());
+  auto logFile = QFile(pathString);
+  if (!logFile.open(QIODevice::ReadOnly | QIODevice::Text)) return nullptr;
+
+  // To save the last report statistics
+  auto stats = LinesData{};
+  auto levels = LinesData{};
+
+  QTextStream in(&logFile);
+  QString line;
+  while (in.readLineInto(&line)) {
+    auto dataLine = line.simplified();
+    if (dataLine.contains(QString(STAT_STR))) {
```

Comment:
These two checks for `STAT_STR` and `DE_STR` might be a bit more efficient if you check for (DE_STR) first and then then do the `STAT_STR` check in an `else if` so we aren't double searching each line and we're cancelling out as early as possible if we find `DE_STR` at the beginning of a line.

If we see slow down from the `STAT_STR` check, assuming `STAT_STR` always starts at the same char position on a line, you might be able to substring a hardcoded portion of the line and compare against that... assuming substring is faster than searching the whole line.

Reply:
@skyler-rs can we mark this as resolved ?